docs/adr/0006_component_constructors.md

```diff
@@ -0,0 +1,334 @@
+# Next Generation Component Constructor Support
+
+* Status: proposed
+* Deciders: Gergely Brautigam, Fabian Burth, Jakob Moeller
+* Date: 2025.05.15
+
+Technical Story: Enable easy and simple construction of component versions based on a `component-constructor.yaml` specification that is backwards compatible and integrates seamlessly with the new OCM Plugin System.
+
+## Context and Problem Statement
+
+Whenever users of OCM want to build component versions, it is important to understand that the existing binding libraries offer support to create component versions with API Calls.
+
+However, when working within the context of a CLI, the user experience is not as straightforward. There is a need for a more user-friendly approach to constructing component versions, especially for those who may not be familiar with the underlying API calls in the Go API directly.
+
+As such we want to continue supporting commands in the form of
+
+```shell
+ocm add componentversions ./ctf component-constructor.yaml
```

Comment:
I wonder if we should rename that to `ocm create componentversions` to make it even more straight-forward.

Reply:
I think we could create an alias but its not super straight forward because not every add is equivalent to create. WDYT?

Reply:
Gotcha. It's not that important. :)